docs/src/index.md

```diff
@@ -131,6 +131,8 @@ julia> gradient(colordiff, RGB(1, 0, 0), RGB(0, 1, 0))
 
 ## Gradients of ML models
 
+### Explicit parameters
```

Comment:
Do we really need 3 levels of headings? The Zygote docs are not very long, and elsewhere 2 is enough. 

If some section needs to be broken into two, that can be done. But this section seems fairly coherent to me, it describes two ways to deal with gradients of the same example.

Is the problem just that "implicit" isn't in a tile? Should the section title be something like "## Explicit and implicit parameters for ML models"?

Reply:
Yeah, the subheadings are redundant if "implicit" and "explicit" are already mentioned in the text.

Reply:
Good point, to me it is a matter of clearly conveying the message that there are 2 main ways of passing parameters. I have nothing for or against 2 vs 3 levels of headings, so I can easily add both types of parameters in the main title as you suggested. When you're browsing through a ton of docs it is helpful to have the relevant information in the headings. 

All these comments are from a perspective of a new user coming without any previous knowledge, so my view on this will certainly differ from yours.

Reply:
Of course, I think @mcabbott's point was that this page is _so_ short that additional headings have a marginal benefit at best. Perhaps it would be better to have the first sentence of each paragraph use the keyword explicit/implicit and italicize it like how implicit is in the current version.

Reply:
Yes I'm sorry if the nit-picking comes off wrong. It's super-valuable to have feedback as to what was hard to understand, and this can be hard for doc-writers to figure out since they are the ones who do already understand. 

The alchemy is in finding ways to address what the problems were, without upsetting what was good before. There's some value to being concise, and to keeping the structure simple.